backend/tests/unit/integrations/llm/test_llm_client.py

```
@@ -809,18 +853,30 @@ async def test_retry_exhausted_on_rate_limit(self, client_with_retry, mock_llm_c
     async def test_empty_response_retry(self, client_with_retry, mock_llm_client):
         """Test retry logic for empty responses."""
         # First response empty, second has content
-        empty_response = Mock()
-        empty_response.content = ""
-        success_response = Mock()
-        success_response.content = "Success after empty retry"
-        
-        mock_llm_client.ainvoke.side_effect = [empty_response, success_response]
+        call_count = [0]
+        
+        def side_effect_func(*args, **kwargs):  # Regular function, not async!
```

Ruff (0.14.0)

Unused function argument: `args`

(ARG001)